source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -2174,6 +2211,29 @@ duplicate (or stale) State Change Errors can be identified (changes in
    the error's topologyVersion (tv2) is not greater than the current
    ServerDescription (tv2).**
 
+Why mark a server Unknown after an auth error?
+''''''''''''''''''''''''''''''''''''''''''''''
+
+The `Authentication spec`_ requires that when authentication fails on a server,
+the driver MUST clear the server's connection pool. Clearing the pool without
+marking the server Unknown would leave the pool in the "paused" state while
+the server is still selectable. When auth fails due to invalid credentials,
+marking the server Unknown also serves to rate limit new connections;
+future operations will need to wait for the server to be rediscovered.
+
+Note that authentication may fail for a variety of reasons, for example:
+
+- A network error, or network timeout error may occur.
+- The server may return a `State Change Error`_.
+- The server may return a AuthenticationFailed command error (error code 18)
+  indicating that the provided credentials are invalid.
+
+Does this mean that authentication failures due to invalid credentials will
+manifest as server selection timeout errors? No, authentication errors are
+still returned to the application immediately. A subsequent operation will
+block until the server is rediscovered and immediately attempt
+authentication on a new connection.
```

Comment:
How is the initial client.connect treated differently so that auth errors don't turn into server selection errors? Or is that supposed to go through selection?

If I'm already connected and there is an auth error when trying the insertOne, is this correct?
```ts
try { await coll.insertOne({ a: 1 }) } catch (e) { e === AuthenticationError }
// Servers are marked as unknown
try { await coll.findOne({ a: 1 }) } catch (e) { e === ServerSelectionError }
```

Reply:
If the credentials are totally invalid, every operation will fail with an auth error (as before) expect that each failure will also mark the server unknown. The behavior will be:
```js
try { await coll.insertOne({ a: 1 }) } catch (e) { e === AuthenticationError }
// Servers are marked as unknown
// Next operation waits for Monitor to rediscover the server
try { await coll.insertOne({ a: 1 }) } catch (e) { e === AuthenticationError }
```

Reply:
Oh I see now, selection will run with that second op, since we have unknowns we wait for the monitor, but when it does attempt a connection we will get the Auth error.